### PR TITLE
Protected route

### PR DESCRIPTION
**Pull request by Gamey001**
Created a protected route component

### What this PR does:
- Stops users from accessing private pages unless they have been authenticated by the system

### Screenshots / Videos:
- 
protectedRoute

### Any information needed to test this feature:
-The Dashboard route was used to Test this feature, try to visit the '/dashboard' route
-Pen the console and check whether the useEffect(ComponentDidMount) hook of the private component has ran
-A user is logged in by default, you can change the login status in the protected route component

### Any issues with the current functionality (optional):
- The protected page renders for loggedIn users. However, a user gets redirected to the login page on failed connection

### CODE REVIEW

File: client/src/components/ProtectedRoute/ProtectedRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { Route, Redirect } from 'react-router-dom';
+type Props = {
+  children: JSX.Element;
+  path: string;
+};
+const ProtectedRoute: React.FC<Props> = ({ children, ...restOfProps }) => {
+  return <Route {...restOfProps} render={() => (true ? children : <Redirect to="/login" />)} />;
```
Comment (rajivtitus) [inline client/src/components/ProtectedRoute/ProtectedRoute.tsx:9]: Maybe check if the user is logged and only then render the component else redirect to Login?

File: client/src/App.tsx [hunk 2]
```diff
@@ -20,12 +20,9 @@ function App(): JSX.Element {
               <Switch>
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
+                <ProtectedRoute path="/dashboard">
```
Comment (rajivtitus) [inline client/src/App.tsx:23]: Just a suggestion, wouldn't it be better to use the exact param to disable partial matching for a route? :)

Comment (Gamey001) [reply client/src/App.tsx:23]: I will do that boss. Thanks!

Comment (Gamey001) [reply client/src/App.tsx]: Both corrections have been effected and await for review @rajiv

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes

**Pull request merged**